src/coprocessor/dag/dag.rs

```diff
@@ -113,6 +112,89 @@ impl DAGContext {
         }
     }
 
+    pub fn handle_streaming_request(&mut self, batch_row_limit: usize) -> Result<Option<Response>> {
+        let mut record_cnt = 0;
+        let mut chunk = Chunk::new();
+        let mut start_key = None;
+        while record_cnt < batch_row_limit {
+            match self.exec.next() {
+                Ok(Some(row)) => {
+                    self.req_ctx.check_if_outdated()?;
+                    record_cnt += 1;
+                    if start_key.is_none() {
+                        start_key = self.exec.take_last_key().or_else(|| Some(vec![]));
+                    }
+                    if self.has_aggr {
+                        chunk.mut_rows_data().extend_from_slice(&row.data.value);
+                    } else {
+                        let value = inflate_cols(&row, &self.columns, &self.output_offsets)?;
+                        chunk.mut_rows_data().extend_from_slice(&value);
+                    }
+                }
+                Ok(None) => {
+                    self.finished = true;
+                    break;
+                }
+                Err(e) => {
+                    self.finished = true;
+                    if let Error::Other(_) = e {
+                        let mut resp = Response::new();
+                        let mut stream_resp = StreamResponse::new();
+                        stream_resp.set_error(to_pb_error(&e));
+                        resp.set_data(box_try!(stream_resp.write_to_bytes()));
+                        resp.set_other_error(format!("{}", e));
+                        return Ok(Some(resp));
+                    } else {
+                        return Err(e);
+                    }
+                }
+            }
+        }
+        if record_cnt == 0 {
+            return Ok(None);
+        }
+        start_key = start_key.and_then(|k| if k.is_empty() { None } else { Some(k) });
+        let end_key = self.exec.take_last_key();
+        DAGContext::make_stream_response(chunk, start_key, end_key).map(Some)
```

Comment:
Should we set output count in each stream response? @lamxTyler

Reply:
Yes.